src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+/**
+ * Task with TaskType, description of task and boolean to check whether the task is done yet.
+ */
+public class Task {
+    private TaskType taskType;
+    private String name;
+    private boolean isMarked;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param taskType Type of task
+     * @param name Description of task
+     * @param isMarked Denotes whether the task is done yet
+     */
+    public Task(TaskType taskType, String name, boolean isMarked) {
+        this.taskType = taskType;
+        this.name = name;
+        this.isMarked = isMarked;
+    }
+
+    /**
+     * @return TaskType of task
+     */
+    public TaskType getTaskType() {
+        return taskType;
+    }
+
+    /**
+     * @return Name of task
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @return Boolean isMarked of task
+     */
+    public boolean isMarked() {
+        return isMarked;
+    }
+
+    /**
+     * Mark the task as done
```

Comment:
(This is just a nitpick but) you might want to rephrase the javadoc header comments in the style "marks", "unmarks", etc. to comply with the coding standard document

Reply:
Yea thanks for reminding and spotted the mistakes in my code!